src/dotnet/commands/dotnet-msbuild/MSBuildForwardingApp.cs

```diff
@@ -32,20 +32,21 @@ public int Execute()
             return new Dictionary<string, string>
             {
                 { "MSBuildExtensionsPath", AppContext.BaseDirectory },
-                { "DotnetHostPath", GetHostPath() },
+                { "CscToolExe", GetRunCscPath() }
             };
         }
 
-        private static string GetHostPath()
-        {
-            return new Muxer().MuxerPath;
-        }
-
         private static string GetMSBuildExePath()
         {
             return Path.Combine(
                 AppContext.BaseDirectory,
                 s_msbuildExeName);
         }
+
+        private static string GetRunCscPath()
+        {
+            var scriptExtension = RuntimeEnvironment.OperatingSystemPlatform == Platform.Windows ? ".cmd" : ".sh";
```

Comment:
I think this should use `RuntimeInformation.IsOSPlatform(OSPlatform.Windows)`, since that is the CoreFX API for detecting the current OS.